jetson/nucleo_bridge/README.md

```diff
@@ -0,0 +1,104 @@
+(e.g. A virtual RA Controller will never attempt to communicate with its physical RA controller unless an RA-related LCM message is sent. This is to prevent multiple virtual Controller objects from trying to contact the same physical Controller object.)
+
+LCMHandler.h is responsible for handling incoming and outgoing lcm messages. \
+Incoming lcm messages will trigger functions which call the functions on the appropriate virtual Controllers. \
+Outgoing lcm messages are triggered by a clock, which query the functions on the appropriate virtual Controllers for data.
+
+I2C.h is responsible for translating communications by virtual Controllers into i2c transactions understood by the linux drivers.
+
+### LCM Channels
+#### RA Open Loop \[Subscriber\] "/ra_openloop_cmd"
+Message: [RAOpenLoopCmd.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/RAOpenLoopCmd.lcm) \
+Publisher: onboard/teleop \
+Subscriber: onboard/nucleo_bridge
+
+#### SA Open Loop \[Subscriber\] "/sa_openloop_cmd"
+Message: [SAOpenLoopCmd.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/SAOpenLoopCmd.lcm) \
+Publisher: onboard/teleop \
+Subscriber: onboard/nucleo_bridge
+
+#### RA Closed Loop \[Subscriber\] "/ik_ra_control"
+Message: [ArmPosition.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/ArmPosition.lcm) \
+Publisher: onboard/kinematics \
+Subscriber: onboard/nucleo_bridge
+
+#### SA Closed Loop \[Subscriber\] "/sa_closedloop_cmd"
+Message: [SAClosedLoopCmd.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/SAClosedLoopCmd.lcm) \
+Publisher: onboard/kinematics \
+Subscriber: onboard/nucleo_bridge
+
+#### Gimbal Open Loop \[Subscriber\] "/gimbal_openloop_cmd"
+Message: [GimbalCmd.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/GimbalCmd.lcm) \
+Publisher: onboard/teleop \
+Subscriber: onboard/nucleo_bridge
+
+#### Hand Open Loop \[Subscriber\] "/hand_openloop_cmd"
+Message: [HandCmd.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/HandCmd.lcm) \
+Publisher: onboard/teleop \
+Subscriber: onboard/nucleo_bridge
+
+#### Foot Open Loop \[Subscriber\] "/foot_openloop_cmd"
+Message: [FootCmd.lcm](https://github.com/raytitan/mrover-workspace/blob/rnucleo/rover_msgs/FootCmd.lcm) \
+Publisher: onboard/teleop \
+Subscriber: onboard/nucleo_bridge
+
+#### RA Pos Data \[Publisher\] "/arm_posiiton"
+Message: [ArmPosition.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/ArmPosition.lcm) \
+Publisher: onboard/nucleo_bridge \
+Subscriber: onboard/kinematics
+
+#### SA Pos Data \[Publisher\] "/sa_pos_data"
+Message: [SAPosData.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/SAPosData.lcm) \
+Publisher: onboard/nucleo_bridge \
+Subscriber: onboard/kinematics
+

```

Comment:
does this code have a watchdog timeout period? or is it on the firmware end? regardless it should documented in the readme

Reply:
There is no watchdog on this one, only in the firmware. It can be added later. Fixed README